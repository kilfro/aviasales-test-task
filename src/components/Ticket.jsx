import '../style/ticket.css'

import FlightSegment from './FlightSegment'
import React from 'react'

const Ticket = ({ price, carrier, segments }) => {
    return (
        <div className='ticket-card'>
            <div className='ticket-header'>
                <div>{price.toLocaleString('ru-ru')} P</div>
                <img src={`http://pics.avs.io/99/36/${carrier}.png`} />
            </div>
            {segments.map(segment => <FlightSegment key={segment.date} {...segment} />)}
        </div>
    )
}

export default Ticket