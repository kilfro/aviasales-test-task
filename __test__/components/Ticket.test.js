import React from 'react'
import Ticket from '../../src/components/Ticket'

describe(Ticket, () => {
  test('must be rendered correctly', () => {
    const props = {
      price: 43073,
      carrier: 'TG',
      segments: [
        {
          origin: 'MOW',
          destination: 'HKT',
          date: '2021-04-03T08:09:00.000Z',
          stops: ['IST', 'BKK', 'DXB'],
          duration: 855,
        },
        {
          origin: 'HKT',
          destination: 'MOW',
          date: '2021-04-22T23:47:00.000Z',
          stops: [],
          duration: 1750,
        },
      ],
    }
    const component = shallow(<Ticket {...props} />)

    expect(component).toMatchSnapshot()
  })
})
