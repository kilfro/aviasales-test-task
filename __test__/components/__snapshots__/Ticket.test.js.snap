// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket must be rendered correctly 1`] = `
<div
  className="ticket-card"
>
  <div
    className="ticket-header"
  >
    <div>
      43Â 073
       P
    </div>
    <img
      src="http://pics.avs.io/99/36/TG.png"
    />
  </div>
  <FlightSegment
    date="2021-04-03T08:09:00.000Z"
    destination="HKT"
    duration={855}
    key="2021-04-03T08:09:00.000Z"
    origin="MOW"
    stops={
      Array [
        "IST",
        "BKK",
        "DXB",
      ]
    }
  />
  <FlightSegment
    date="2021-04-22T23:47:00.000Z"
    destination="MOW"
    duration={1750}
    key="2021-04-22T23:47:00.000Z"
    origin="HKT"
    stops={Array []}
  />
</div>
`;
